flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/api/functions/source/MessageAcknowledingSourceBase.java

```diff
@@ -144,6 +152,20 @@ protected void addId(Id id) {
 	
 	@Override
 	public SerializedCheckpointData[] snapshotState(long checkpointId, long checkpointTimestamp) throws Exception {
+		if (idsForCurrentCheckpoint == null) {
```

Comment:
I think these checks are pretty redundant. The tasks should guarantee that no calls happen on non-open functions, or closed functions.